src/gui/clips/MidiClipView.cpp

```diff
@@ -144,8 +146,66 @@ void MidiClipView::changeName()
 
 
 
+void MidiClipView::transposeSelection()
+{
+	const auto selection = getClickedClips();
+
+	// Calculate the key boundries for all clips
+	int highest = 0;
+	int lowest = NumKeys - 1;
+	for (ClipView* clipview: selection)
+	{
+		if (auto mcv = dynamic_cast<MidiClipView*>(clipview))
```

Comment:
`qobject_cast` would be more efficient here (and further down in this file).

Reply:
Thanks! I'll add it for the sake of good practice.